### PR TITLE
Add ONNX node tests to CI, cleanup more pure ops

### PR DESCRIPTION
**Pull request by orausch**
With the new decorator feature, we can simplify a whole bunch of ops.

Also added the test cases ONNX ships with to our CI, this is quite
useful for development. For now, we are still skipping quite a few,
especially on the more complex ops.

### CODE REVIEW

File: daceml/onnx/op_implementations/pure_implementations.py [hunk 1]
```diff
@@ -882,18 +738,9 @@ def Softplus(X, Y):
...
-        dtype = in_desc_with_name(node, state, sdfg, "X").dtype
-
-        def prog(X, Y):
-            Y[:] = dace.elementwise(lambda x: dtype(1) / (dtype(1) + exp(-x)),
-                                    X)
-
-        return program_for_node(prog, sdfg, state, node)
+@python_pure_op_implementation(dtype=lambda X: X.dtype)
+def Sigmoid(X, Y):
+    Y[:] = dace.elementwise(lambda x: dtype(1) / (dtype(1) + exp(-x)), X)
```
Comment (tbennun) [inline daceml/onnx/op_implementations/pure_implementations.py:743]: for the other ones you used a string. Why did you not use one here?

File: setup.py [hunk 2]
```diff
@@ -37,8 +37,11 @@
         exclude=["*.tests", "*.tests.*", "tests.*", "tests"]),
     package_data={'': (['*.cpp'] + runtime_files)},
     install_requires=[
-        'dace@git+https://github.com/spcl/dace.git@40d735', 'onnx == 1.8.0',
-        'torch', 'protobuf == 3.19', 'dataclasses; python_version < "3.7"',
+        'dace == 0.13.3',
+        'onnx == 1.7.0',  # we support opset v12
```
Comment (tbennun) [inline setup.py:41]: did we downgrade ONNX here?

**Review by tbennun — APPROVED** [review]
Please answer comments, otherwise LGTM! Some nice engineering here too.

File: daceml/onnx/op_implementations/pure_implementations.py (hunk 1 above)
Comment (orausch) [reply daceml/onnx/op_implementations/pure_implementations.py:743]: I don't remember, there must have been some bug back when I wrote it

File: setup.py (hunk 2 above)
Comment (orausch) [reply setup.py:41]: ONNX 1.8's built in tests are the tests for opset 13, but we implement opset 12

**Pull request merged**